src/aoapplication.cpp

```diff
@@ -232,17 +234,9 @@ QString AOApplication::get_current_char()
  * @return A sanitized path. If any check fails, the path returned is an empty string. The sanitized path does not
  * necessarily exist.
  */
-QString AOApplication::sanitize_path(QString p_file)
+bool AOApplication::is_safe_path(QString p_file)
 {
-  if (!p_file.contains(".."))
-    return p_file;
-
-  QStringList list = p_file.split(QRegularExpression("[\\/]"));
-  while (!list.isEmpty())
-    if (list.takeFirst().contains(QRegularExpression("\\.{2,}")))
-      return nullptr;
-
-  return p_file;
+  return !p_file.contains("..");
```

Comment:
A few vanilla files include .. as part of the name (e.g. "base/sounds/music/Desire for Execution....mp3"), so this wouldn't work